java-symbolic-execution/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java

```diff
@@ -407,12 +407,47 @@ private void handleLiteral(LiteralTree tree) {
       if (tree.is(Tree.Kind.CHAR_LITERAL) && isNullCharacter(value)) {
         addZeroConstraint(sv, ZeroConstraint.ZERO);
       } else if (tree.is(Tree.Kind.INT_LITERAL, Tree.Kind.LONG_LITERAL, Tree.Kind.DOUBLE_LITERAL, Tree.Kind.FLOAT_LITERAL)) {
-        addZeroConstraint(sv, isNumberZero(value) ? ZeroConstraint.ZERO : ZeroConstraint.NON_ZERO);
+        addZeroConstraint(sv, isZeroLiteral(tree) ? ZeroConstraint.ZERO : ZeroConstraint.NON_ZERO);
       }
     }
 
-    private static boolean isNumberZero(String literalValue) {
-      return !(literalValue.matches("(.)*[1-9]+(.)*") || literalValue.matches("(0x|0X){1}(.)*[1-9a-fA-F]+(.)*") || literalValue.matches("(0b|0B){1}(.)*[1]+(.)*"));
+    // LiteralTree.asConstant() could fit here logically but is not an optimal algorithm for such a hot paths
```

Comment:
Worth writing it in a proper javadoc comment, and give a bit more of the why.

But I'm now wondering, what about caching the couples (literal tree value; boolean)? That way we would not have to recompute it all the time as well.